application/ui/src/features/inspect/inference-provider.component.tsx

```diff
@@ -64,14 +65,25 @@ export const InferenceProvider = ({ children }: InferenceProviderProps) => {
     const [selectedModelId, setSelectedModelId] = useState<string | undefined>(undefined);
     const [inferenceOpacity, setInferenceOpacity] = useState<number>(0.75);
 
+    const { selectedMediaItem } = useSelectedMediaItem();
+
+    const onSetSelectedModelId = (modelId: string | undefined) => {
+        setSelectedModelId(modelId);
+
+        if (modelId) {
+            if (selectedMediaItem) {
```

Comment:
Any particular reason for having them in separate if statements instead of combining them into one?